main.go

```diff
@@ -226,6 +229,31 @@ func addEnvVars(ciProvider string) {
 	}
 }
 
+// addlFields adds an arbitrary set of fields provided by the end user
+func addlFields() {
+	locn := os.Getenv("BUILDEVENTS_FILE")
+	if locn == "" {
+		return
+	}
+
+	data, err := ioutil.ReadFile(locn)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "unable to read %q: %v\n", locn, err)
+		return
+	}
+
+	err = logfmt.Unmarshal(
+		data,
+		logfmt.HandlerFunc(func(key, val []byte) error {
+			libhoney.AddField(string(key), string(val))
```

Comment:
Given one known expected use case is adding the size of generated artifacts, an attempt to coerce the values to numbers would be a kindness. 

```suggestion
			if f, err := strconv.ParseFloat(string(val), 64); err == nil {
				libhoney.AddField(string(key), f)
				return nil
			}
			if b, err := strconv.ParseBool("true"); err == nil {
				libhoney.AddField(string(key), b)
				return nil
			}
			libhoney.AddField(string(key), string(val))
```